### PR TITLE
feat(wowza streaming cloud):

### PR DESCRIPTION
**Pull request by DiLiUA**
- updated dependencies to angulart 5 and ngx-bootstrap 2.0.0-beta.8
- changed styleExt to SCSS in angular cli
- added wowza streaming cloud;
- added my-events page;
- added my-event component
- validated fields Lounch page
- added time picker, back btn Lounch page
- updated date picker

### CODE REVIEW

File: src/event-launch/event-launch.component.html [hunk 1]
```diff
@@ -1,210 +1,303 @@
 <div class="launch-event-container">
   <div class="container">
-    <div class="first-step" *ngIf="!secondStepActive">
+    <div class="first-step" *ngIf="activeStep === 1">
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.html]: avoid direct comparison in html

File: src/event-launch/event-launch.component.html [hunk 2]
```diff
@@ -1,210 +1,303 @@
...
-                  </div>
-                </div>
+                <ul *dropdownMenu class="dropdown-menu drodown-custom-scroll" role="menu">
+                  <li role="menuitem" *ngFor="let location of locations" class="dropdown-item">
+                    <label [for]="location.id">{{location.name}}</label>
+                    <input type="radio"
+                           name="showLocation"
+                           [(ngModel)]="launchEvent.showLocation"
+                           #showLocation="ngModel"
+                           [id]="location.id"
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.html:75]: check if you can use  id="location.id" without [ ]

File: src/event-launch/event-launch.component.html [hunk 3]
```diff
@@ -1,210 +1,303 @@
...
+                           [value]="location.name"
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.html:76]: the same as above

File: src/event-launch/event-launch.component.html [hunk 4]
```diff
@@ -1,210 +1,303 @@
...
+            <div class="form-group">
+              <label for="show-tickets">How many people would join me:</label>
+              <input type="number"
+                     id="show-tickets"
+                     class="form-control"
+                     placeholder="min 50"
+                     name="tickets"
+                     [(ngModel)]="launchEvent.tickets.count"
+                     #tickets="ngModel"
+                     [min]="50"
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.html]: avoid use dual-binding [ ]

File: src/event-launch/event-launch.component.html [hunk 5]
```diff
@@ -1,210 +1,303 @@
...
+            <div class="form-group">
+              <label for="price">They willing to pay per ticket:</label>
+              <input type="number"
+                     id="price"
+                     class="form-control"
+                     placeholder="min $10"
+                     name="ticketPrice"
+                     [(ngModel)]="launchEvent.tickets.ticketPrice"
+                     #ticketPrice="ngModel"
+                     [min]="10"
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.html]: min="10"

File: src/event-launch/event-launch.component.html [hunk 6]
```diff
@@ -1,210 +1,303 @@
...
+            <div class="form-group">
+              <label for="ticketsSold">Amount of sold tickets to perform the show 0:</label>
+              <input type="number"
+                     id="ticketsSold"
+                     class="form-control"
+                     placeholder="A critical minimum of tickets to be sold"
+                     name="ticketsSold"
+                     [(ngModel)]="launchEvent.tickets.ticketsToFund"
+                     #ticketsSold="ngModel"
+                     [min]="0"
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.html]: min="0"

File: src/event-launch/event-launch.component.html [hunk 7]
```diff
@@ -1,210 +1,303 @@
...
 
-        <div class="description-container">
-          <label for="decription" class="input-label">Description</label>
-          <textarea name="decription"
-                    id="decription"
-                    [(ngModel)]="launchEvent.description"
-                    placeholder="Write a show description here">
-          </textarea>
-        </div>
+    <div class="second-step" *ngIf="activeStep === 2">
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.html]: avoid using direct comparison in html

File: src/event-launch/event-launch.component.html [hunk 8]
```diff
@@ -1,210 +1,303 @@
...
+      <form class="row" #eventFormStep2="ngForm">
+        <div class="col-lg-9 col-md-9 col-sm-9 col-xs-9">
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.html]: use col-9

File: src/event-launch/event-launch.component.html [hunk 9]
```diff
@@ -1,210 +1,303 @@
...
+            <label>Add some info:</label>
+            <button class="add-button">Add +</button>
           </div>
 
-          <div class="artists-appriciations">
+          <app-list-of-encoders (setEncoder)="setEncoder($event)"></app-list-of-encoders>
+
+          <button type="button" class="next-button" (click)="goBack()">< Previous step</button>
+        </div>
+        <div class="side-bar col-lg-3 col-md-3 col-sm-3 col-xs-3">
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.html]: use col-3

File: src/event-launch/event-launch.component.ts [hunk 10]
```diff
@@ -75,76 +61,126 @@ export class LaunchComponent implements OnInit {
...
       live: false,
-      appreciations: {}
+      appreciations: {},
+      wowza: {
+        id: null
+      }
     };
 
     this.searchServiceSubscribe = this.searchService.getMusicStyles()
       .subscribe((res: any): void => {
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.ts]: avoid **any**

File: src/event-launch/event-launch.component.ts [hunk 11]
```diff
@@ -75,76 +61,126 @@ export class LaunchComponent implements OnInit {
...
-        if (res.error) {
-          console.error(res.error);
-          return;
-        }
         const styles: any = res.data;
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.ts]: avoid **any**

File: src/event-launch/event-launch.component.ts [hunk 12]
```diff
@@ -75,76 +61,126 @@ export class LaunchComponent implements OnInit {
...
-        this.genres = styles.genres;
+        this.genres = styles.genres.map((genre: string) => ({isChecked: false, value: genre}));
+      }, err => {
+        console.error(err);
       });
 
     this.searchServiceSubscribe = this.searchService.getLocations()
       .subscribe((res: any): void => {
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.ts]: avoid **any**

File: src/event-launch/event-launch.component.ts [hunk 13]
```diff
@@ -75,76 +61,126 @@ export class LaunchComponent implements OnInit {
...
+  }
+
+  goToNextStep(eventForm: NgForm): void {
+    this.launchEvent.genres = this.genres.filter(genre => genre.isChecked).map(genre => genre.value);
+
+    if (!eventForm.invalid && this.launchEvent.genres.length) {
+      this.launchEvent.creator = this.userProfile.email;
+      this.secondStepActive = true;
+      this.activeStep += 1;
+    } else {
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.ts]: avoid **else**

File: src/event-launch/event-launch.component.ts [hunk 14]
```diff
@@ -75,76 +61,126 @@ export class LaunchComponent implements OnInit {
...
+  ngOnDestroy() {
 
-    if (index !== -1) {
-      this.selectedGenres.splice(index, 1);
+  }
+
+  goToNextStep(eventForm: NgForm): void {
+    this.launchEvent.genres = this.genres.filter(genre => genre.isChecked).map(genre => genre.value);
+
+    if (!eventForm.invalid && this.launchEvent.genres.length) {
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.ts]: use negative scenario at first

File: src/event-launch/event-launch.component.ts [hunk 15]
```diff
@@ -75,76 +61,126 @@ export class LaunchComponent implements OnInit {
...
       return;
     }
 
-    this.selectedGenres.push(genre);
+    this.wowzaObj.name = this.launchEvent.showName;
+    this.launchEvent.timePerfomance.start = moment(this.timePerfomance.start).format('dddd, MMMM DD YYYY, h:mm:ss a');
+    this.launchEvent.timePerfomance.end = moment(this.timePerfomance.end).format('dddd, MMMM DD YYYY, h:mm:ss a');
+
+    const newStreamSubscription = this.wowzaCloudService.newStream(this.wowzaObj)
+      .subscribe((liveStream: any) => {
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.ts]: use **liveStream => **

File: src/event-launch/event-launch.component.ts [hunk 16]
```diff
@@ -75,76 +61,126 @@ export class LaunchComponent implements OnInit {
...
+        this.launchEvent.wowza.id = liveStream.live_stream.id;
+        this.eventServiceSubscribe = this.eventService.saveNewEvent(this.launchEvent)
+          .subscribe((res: any) => {
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.ts]: use **res =>**

File: src/event-launch/event-launch.component.ts [hunk 17]
```diff
@@ -75,76 +61,126 @@ export class LaunchComponent implements OnInit {
...
+            this.router.navigate(['events']);
+          }, (err: any) => {
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.ts]: use err=>

File: src/event-launch/event-launch.component.ts [hunk 18]
```diff
@@ -75,76 +61,126 @@ export class LaunchComponent implements OnInit {
...
+            console.log(err);
+            return;
+          });
+      }, (err: any) => {
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.ts]: the same as above

File: src/shared/search-service/search.service.ts [hunk 19]
```diff
@@ -22,6 +22,7 @@ export class SearchService {
   }
 
   public getLocations(): Observable<any> {
+
     return this.http.get(`${Config.api}/get-locations`).map((res: any) => {
```
Comment (Ugmaxie) [inline src/shared/search-service/search.service.ts:26]: res =>

File: src/shared/wowza-streaming-cloud/wowza-cloud.component.ts [hunk 20]
```diff
@@ -0,0 +1,125 @@
...
+        this.showData = {
+          ...params,
+          wowza: {
+            ...stream,
+            ...{player_hls_playback_url: 'https://10e8f0.entrypoint.cloud.wowza.com/app-8de5/ngrp:03bae8e8_all/playlist.m3u8'}
+          }
+        };
+
+        const streamStateSubsciption = this.wowzaCloudService.getStreamState(this.showData.wowza.id)
+          .subscribe((state: string) => {
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-cloud.component.ts]: state =>

File: src/shared/wowza-streaming-cloud/wowza-cloud.component.ts [hunk 21]
```diff
@@ -0,0 +1,125 @@
...
+            console.log('state', state);
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-cloud.component.ts]: avoid console.log

File: src/shared/wowza-streaming-cloud/wowza-cloud.component.ts [hunk 22]
```diff
@@ -0,0 +1,125 @@
...
+            this.isStartedStream = state === 'started' || state === 'starting';
+          }, err => {
+            console.log('err: ', JSON.parse(err._body));
+          });
+
+        this.subManager.add(streamStateSubsciption);
+
+        console.log('showData', this.showData.wowza);
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-cloud.component.ts]: avoid console.log

File: src/shared/wowza-streaming-cloud/wowza-cloud.component.ts [hunk 23]
```diff
@@ -0,0 +1,125 @@
...
+      }, err => {
+        console.log('err: ', JSON.parse(err._body));
+      });
+
+    this.subManager.add(streamSubscription);
+  };
+
+  showData: any;
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-cloud.component.ts]: avoid any

File: src/shared/wowza-streaming-cloud/wowza-cloud.component.ts [hunk 24]
```diff
@@ -0,0 +1,125 @@
...
+          console.log('err start stream', err);
+        });
+
+      this.subManager.add(startSubscription);
+    }
+  }
+
+  regenerateConnectionCode() {
+    const codeSubscription = this.wowzaCloudService.regenerateConnectionCode(this.showData.wowza.id)
+      .subscribe((res: any) => {
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-cloud.component.ts]: res =>

File: src/shared/wowza-streaming-cloud/wowza-cloud.component.ts [hunk 25]
```diff
@@ -0,0 +1,125 @@
...
+        this.showData.wowza.connection_code = res;
+      }, (err: any) => {
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-cloud.component.ts]: err =>

File: src/shared/wowza-streaming-cloud/wowza-cloud.component.ts [hunk 26]
```diff
@@ -0,0 +1,125 @@
...
+    this.subManager.add(codeSubscription);
+  }
+
+  stopStream() {
+    if (!this.isStartedStream) {
+      return;
+    }
+
+    const stopSubscription = this.wowzaCloudService.stopStream(this.showData.wowza.id)
+      .subscribe((res: any) => {
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-cloud.component.ts]: res =>

File: src/shared/wowza-streaming-cloud/wowza-cloud.component.ts [hunk 27]
```diff
@@ -0,0 +1,125 @@
...
+        console.log('stream stopped', res);
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-cloud.component.ts]: console.log

File: src/shared/wowza-streaming-cloud/wowza-cloud.component.ts [hunk 28]
```diff
@@ -0,0 +1,125 @@
...
+        this.isStartedStream = false;
+      }, (err: any) => {
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-cloud.component.ts]: err =>

File: src/shared/wowza-streaming-cloud/wowza-cloud.component.ts [hunk 29]
```diff
@@ -0,0 +1,125 @@
...
+        console.log('err stop stream', err);
+      });
+
+    this.subManager.add(stopSubscription);
+
+  }
+
+  changeEncoder(encoder: string) {
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-cloud.component.ts]: : void

File: src/shared/wowza-streaming-cloud/wowza-cloud.component.ts [hunk 30]
```diff
@@ -0,0 +1,125 @@
...
+    const codeSubscription = this.wowzaCloudService.updateStream(this.showData.wowza.id, {encoder})
+      .subscribe((res: any) => {
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-cloud.component.ts]: res =>

File: src/shared/wowza-streaming-cloud/wowza-cloud.component.ts [hunk 31]
```diff
@@ -0,0 +1,125 @@
...
+        this.showData.wowza = {
+          ...res,
+          ...{
+            player_hls_playback_url: 'https://10e8f0.entrypoint.cloud.wowza.com/app-8de5/ngrp:03bae8e8_all/playlist.m3u8'}
+            // should be deleted after by wowza licence
+        };
+      }, (err: any) => {
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-cloud.component.ts]: err =>

File: src/shared/wowza-streaming-cloud/wowza-cloud.component.ts [hunk 32]
```diff
@@ -0,0 +1,125 @@
...
+        console.log('err start stream', err);
+      });
+
+    this.subManager.add(codeSubscription);
+  }
+
+  newStream() {
+    const newStreamSubscription = this.wowzaCloudService.newStream('')
+      .subscribe((res: any) => {
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-cloud.component.ts]: res =>

File: src/shared/wowza-streaming-cloud/wowza-cloud.component.ts [hunk 33]
```diff
@@ -0,0 +1,125 @@
...
+        console.log('New stream', res);
+      }, (err: any) => {
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-cloud.component.ts]: err =>

File: src/shared/wowza-streaming-cloud/wowza-player/wowza-player.component.ts [hunk 34]
```diff
@@ -0,0 +1,55 @@
...
+    this.showParams = showParams;
+
+    Promise.resolve(null)
+      .then(() => {
+        this.initPlayer();
+      });
+
+  }
+
+  showParams: any;
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-player/wowza-player.component.ts]: avoid any

File: src/shared/wowza-streaming-cloud/wowza-player/wowza-player.component.ts [hunk 35]
```diff
@@ -0,0 +1,55 @@
...
+  player: any;
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-player/wowza-player.component.ts]: avoid any

File: src/shared/wowza-streaming-cloud/wowza-player/wowza-player.component.ts [hunk 36]
```diff
@@ -0,0 +1,55 @@
...
+
+  constructor(private wowzaCloudService: WowzaCloudService) {
+
+  }
+
+  initPlayer(): void {
+    this.player = this.wowzaCloudService.initPlayer(this.parsePlayerConfig(), this.playerTemplateId);
+  }
+
+  parsePlayerConfig(): any {
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-player/wowza-player.component.ts]: avoid any

**Review by Ugmaxie — CHANGES_REQUESTED** [review]

File: src/event-launch/event-launch.component.html (hunk 2 above)
Comment (DiLiUA) [reply src/event-launch/event-launch.component.html:75]: I can use this  `id="{{location.id}}"` but I think  that `[id]="location.id"` more beautiful

File: src/event-launch/event-launch.component.html (hunk 3 above)
Comment (DiLiUA) [reply src/event-launch/event-launch.component.html:76]: I can use this  `id="{{location.id}}"` but I think  that `[id]="location.id"` more beautiful

File: src/shared/wowza-streaming-cloud/wowza-player/wowza-player.component.ts (hunk 35 above)
Comment (DiLiUA) [reply src/shared/wowza-streaming-cloud/wowza-player/wowza-player.component.ts]: `player` it's an Object from Wowza Player API. Should I create interface for it?

File: src/shared/search-service/search.service.ts (hunk 19 above)
Comment (DiLiUA) [reply src/shared/search-service/search.service.ts:26]: If I remove type any I have linter Error like `_body does not exist type Response`

**Review by Ugmaxie — APPROVED** [review]

**Pull request merged**